xbmc/windowing/X11/WinSystemX11GLESContext.h

```diff
@@ -0,0 +1,54 @@
+/*
+ *  Copyright (C) 2005-2018 Team Kodi
+ *  This file is part of Kodi - https://kodi.tv
+ *
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *  See LICENSES/README.md for more information.
+ */
+
+#pragma once
+
+#include "WinSystemX11.h"
+
+#include "EGL/egl.h"
+#include "rendering/gles/RenderSystemGLES.h"
+#include "platform/linux/OptionalsReg.h"
+#include <memory>
+
+class CGLContextEGL;
+
+class CWinSystemX11GLESContext : public CWinSystemX11, public CRenderSystemGLES
+{
+public:
+  CWinSystemX11GLESContext();
+  virtual ~CWinSystemX11GLESContext();
+
+  // Implementation of CWinSystem via CWinSystemX11
+  CRenderSystemBase *GetRenderSystem() override { return this; }
+  bool CreateNewWindow(const std::string& name, bool fullScreen, RESOLUTION_INFO& res) override;
+  bool ResizeWindow(int newWidth, int newHeight, int newLeft, int newTop) override;
+  void FinishWindowResize(int newWidth, int newHeight) override;
+  bool SetFullScreen(bool fullScreen, RESOLUTION_INFO& res, bool blankOtherDisplays) override;
+  bool DestroyWindowSystem() override;
+  bool DestroyWindow() override;
+
+  bool IsExtSupported(const char* extension) const override;
+
+  EGLDisplay GetEGLDisplay() const;
+  EGLSurface GetEGLSurface() const;
+  EGLContext GetEGLContext() const;
+  EGLConfig GetEGLConfig() const;
+
+protected:
+  bool SetWindow(int width, int height, bool fullscreen, const std::string &output, int *winstate = NULL) override;
```

Comment:
- left align `&` and `*`, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#52-pointer-and-reference-types
- use nullptr, https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md#104-null-vs-nullptr